main.js

```diff
@@ -0,0 +1,194 @@
+var posts = [];
+var currId = 1;
+
+var Post = function (id, name, message) {
+  this.id = id;
+  this.name = name;
+  this.message = message;
+  this.currCommentId = 1;
+  this.comments = [];
+
+  this.toggleEditForm = function () {
+    var $editForm = $('#' + this.id).find('.edit-post-form');
+    $editForm.find('#edit-' + this.id).attr('value', this.message);
+
+    $('#' + this.id).find('.edit-post-form').toggle();
+  }
+
+  this.toggleComments = function () {
+    var $post = $('#' + this.id);
+
+    $post.siblings().toggle();    
+    $('#post-form, a:not(.show-comments)').toggle();
+    $post.find('.comment-section').toggle();
+
+    var $editForm = $post.find('.edit-post-form');
+
+    if ($editForm.css('display') != 'none')
+      this.toggleEditForm();
+  }
+
+  this.editMessage = function (newMessage) {
+    if (newMessage && newMessage != this.message) {
+      var edit = '<p class="post-message">' + newMessage + '<em> (edited)</em> - ' + '<strong>Posted By:</strong> ' + this.name + '</p>';
+      var $post = $('#' + this.id);
+
+      $post.find('.post-message').remove();
+      $post.prepend(edit);
+      this.message = newMessage;
+
+      $('#edit-form-' + this.id)[0].reset();
+      this.toggleEditForm();
+    }
+  }
+
+  this.newComment = function (comment, commenter, $commentSection) {
+    var newId = this.currCommentId;
+    this.currCommentId++;
+
+    if (comment && commenter) {
+      $commentSection.find('.comments').append(
+        '<div class="comment" id="' + this.id + '-' + newId + '">'
+        + '<hr><p>' + comment + ' - ' + '<strong>Posted By:</strong> ' + commenter + '</p>'
+        + '<a class="remove-comment">remove comment</a>'
+        +'</div>'
+      );
+
+      $('#comment-form-' + this.id)[0].reset();
+
+      this.comments.push(
+        {
+          id: this.id + '-' + newId,
+          commenter: commenter,
+          comment: comment
+        }
+      );
+    }
+  }
+
+  this.removeComment = function (commentId) {
+    var index = this.comments.findIndex(function (comment) {
+      return comment.id === commentId;
+    });
+
+    var $commentDiv = $('#' + commentId);
+
+    this.comments.splice(index, 1);
+    $commentDiv.remove();
+  }
+}
+
+// ----- Returns the corresponding post -----
+var getPost = function (id) {
+  return posts.find(function (post) {
+    return post.id == id;
+  });
+}
+
+// ----- Create new Post -----
+$('#submit').on('click', function () {
+  var message = $('#message').val();
```

Comment:
The risk with chaining a bunch of jQuery methods (`.val()`) is that it makes your code less resilient if the DOM happens to change. It might be good to consider other strategies.